iocage/cli/exec.py

```diff
@@ -24,31 +24,52 @@
 """exec module for the cli."""
 import click
 
-import iocage.lib.ioc_common as ioc_common
-import iocage.lib.iocage as ioc
+import iocage.lib.Jail
+import iocage.lib.Logger
+
 
 __rootcmd__ = True
 
 
-@click.command(context_settings=dict(
-    ignore_unknown_options=True, ),
-    name="exec", help="Run a command inside a specified jail.")
+@click.command(context_settings=dict(ignore_unknown_options=True),
+               name="exec", help="Run a command inside a specified jail.")
 @click.option("--host_user", "-u", default="root",
               help="The host user to use.")
 @click.option("--jail_user", "-U", help="The jail user to use.")
 @click.argument("jail", required=True, nargs=1)
 @click.argument("command", nargs=-1, type=click.UNPROCESSED)
-def cli(command, jail, host_user, jail_user):
+@click.option("--log-level", "-d", default=None)
+def cli(command, jail, host_user, jail_user, log_level):
     """Runs the command given inside the specified jail as the supplied
     user."""
-    # We may be getting ';', '&&' and so forth. Adding the shell for safety.
-    if len(command) == 1:
-        command = ("/bin/sh", "-c") + command
+    logger = iocage.lib.Logger.Logger(print_level=log_level)
 
     if jail.startswith("-"):
-        ioc_common.logit({
-            "level"  : "EXCEPTION",
-            "message": "Please specify a jail first!"
-        })
+        logger.error("Please specify a jail first!")
+        exit(1)
+
+    user_command = " ".join(list(command))
+
+    if jail_user:
+        command = ["/bin/su", "-m",
+                   escape_shell_arg(jail_user), "-c", user_command]
```

Comment:
Is command being run with shell=True? if not i would not see the reason to escape this.